lib/IgnorePlugin.js

```diff
@@ -8,14 +8,20 @@
 
 class IgnorePlugin {
 	/**
-	 * @param {RegExp} resourceRegExp A RegExp to test the request against
-	 * @param {RegExp=} contextRegExp A RegExp to test the context (directory) against
+	 * @param {object} options IgnorePlugin options
+	 * @param {RegExp} options.resourceRegExp - A RegExp to test the request against
+	 * @param {RegExp} options.contextRegExp - A RegExp to test the context (directory) against
+	 * @param {TODO} options.checkResource - A function that will take resource as argument and should return true or false
+	 * @param {TODO} options.checkContext - A function that will take context as argument and should return true or false
 	 */
-	constructor(resourceRegExp, contextRegExp) {
-		/** @private @type {RegExp} */
-		this.resourceRegExp = resourceRegExp;
-		/** @private @type {RegExp} */
-		this.contextRegExp = contextRegExp;
+	constructor(options) {
+		this.options = options;
+		if (options.checkResource) {
```

Comment:
Don't override member functions. That doesn't look right.

Instead add an `if` to the `checkResource` method.

Reply:
This was to avoid thousands or more unnecessary checks.